ledger-api/grpc-definitions/com/daml/ledger/api/v2/reassignment.proto

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+import "com/daml/ledger/api/v1/event.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "ReassignmentOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// Complete view of an on-ledger reassignment.
+message Reassignment {
+
+  // Assigned by the server. Useful for correlating logs.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string update_id = 1;
+
+  // The ID of the command which resulted in this reassignment. Missing for everyone except the submitting party on the submitting participant.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string command_id = 2;
+
+  // The workflow ID used in reassignment command submission. Only set if the ``workflow_id`` for the command was set.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = 3;
+
+  // The absolute offset. The format of this field is described in ``ledger_offset.proto``.
+  // Required
+  string offset = 4;
+
+  oneof event {
+    UnassignedEvent unassigned_event = 5;
+    AssignedEvent assigned_event = 6;
+  }
+}
+
+// Records that a contract has been unassigned, and it becomes unusable on the source domain
+message UnassignedEvent {
+
+  // The ID of the reassignment. This needs to be used as an input for a assign ReassignmentCommand.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string reassignment_id = 1;
```

Comment:
Should we call it `unassign_id` ? I'd like to avoid the confusion with the internal transfer id (reassignment id) which consists of the source domain id and a timestamp. Here, we only have the timestamp.

Reply:
hm, I see, makes sense to me. Added this change, can you have a look again?